src/functional/swing_slide_filter.zig

```diff
@@ -70,20 +70,24 @@ pub fn compressSwing(
     var current_timestamp: usize = 2;
     while (current_timestamp < uncompressed_values.len) : (current_timestamp += 1) {
         // Evaluate the upper and lower bound linear functions at current timestamp.
-        const upper_limit = evaluateLinearFunctionAtTime(upper_bound_linear_function, current_timestamp);
-        const lower_limit = evaluateLinearFunctionAtTime(lower_bound_linear_function, current_timestamp);
+        const upper_limit = evaluateLinearFunctionAtTime(
+            upper_bound_linear_function,
+            @floatFromInt(current_timestamp),
+        );
+        const lower_limit = evaluateLinearFunctionAtTime(
+            lower_bound_linear_function,
+            @floatFromInt(current_timestamp),
+        );
 
         if ((upper_limit < (uncompressed_values[current_timestamp] - adjusted_error_bound)) or
             (lower_limit > (uncompressed_values[current_timestamp] + adjusted_error_bound)))
         {
             // Recording mechanism (the current points is outside the limits).
             try appendValue(f64, current_segment.start_point.value, compressed_values);
-            const segment_size = current_timestamp - current_segment.start_point.time - 1;
+            const segment_size = @as(f64, @floatFromInt(current_timestamp)) - current_segment.start_point.time - 1;
```

Comment:
If continuous time is kept in Swing, maybe extend the doc comment of the module to make it clear why. The use of an `f64` for time in the form of indices as specified in #34 and all of the `@floatFromInt()` casts looks very strange without this information.